src/base/bittorrent/statistics.cpp

```diff
@@ -79,30 +82,34 @@ void Statistics::gather()
         m_dirty = true;
     }
 
-    save();
+    if (m_lastUpdateTimer.hasExpired(SAVE_INTERVAL))
+        save();
 }
 
 void Statistics::save() const
 {
-    const qint64 now = QDateTime::currentSecsSinceEpoch();
-
-    if (!m_dirty || ((now - m_lastWrite) < SAVE_INTERVAL))
+    if (!m_dirty)
         return;
 
-    SettingsPtr s = Profile::instance()->applicationSettings(u"qBittorrent-data"_qs);
-    QVariantHash v;
-    v.insert(u"AlltimeDL"_qs, (m_alltimeDL + m_sessionDL));
-    v.insert(u"AlltimeUL"_qs, (m_alltimeUL + m_sessionUL));
-    s->setValue(u"Stats/AllStats"_qs, v);
+    const QVariantHash stats =
+    {
+        {u"AlltimeDL"_qs, (m_alltimeDL + m_sessionDL)},
+        {u"AlltimeUL"_qs, (m_alltimeUL + m_sessionUL)}
+    };
+    SettingsPtr settings = Profile::instance()->applicationSettings(u"qBittorrent-data"_qs);
+    settings->setValue(u"Stats/AllStats"_qs, stats);
+
+    m_lastUpdateTimer.start();
```

Comment:
I would move it to constructor.

Reply:
>I would move it to constructor.

I don't think that would work correctly, or do you mean line 114 should move to constructor?

Reply:
> or do you mean line 114 should move to constructor?

Oops... Yes!

Reply:
Done.